core/config/toml/toml.go

```diff
@@ -0,0 +1,257 @@
+package toml
+
+import (
+	"net"
+	"net/url"
+
+	ocrnetworking "github.com/smartcontractkit/libocr/networking"
+	"go.uber.org/zap/zapcore"
+
+	"github.com/smartcontractkit/chainlink/core/config"
+	"github.com/smartcontractkit/chainlink/core/services/keystore/keys/ethkey"
+	"github.com/smartcontractkit/chainlink/core/services/keystore/keys/p2pkey"
+	"github.com/smartcontractkit/chainlink/core/store/models"
+	"github.com/smartcontractkit/chainlink/core/utils"
+)
+
+//TODO doc
+type CoreConfig struct {
+	// General/misc
+	Dev                 *bool
+	ExplorerURL         *URL
+	InsecureFastScrypt  *bool
+	ReaperExpiration    *models.Duration
+	RootDir             *string
+	ShutdownGracePeriod *models.Duration
+
+	Database *DatabaseConfig
+
+	TelemetryIngress *TelemetryIngressConfig
+
+	Log *LogConfig
+
+	WebServer *WebServerConfig
+
+	// Feeds manager
+	FeatureFeedsManager *bool
+	FeatureUICSAKeys    *bool
+
+	// LogPoller
+	FeatureLogPoller *bool
+
+	// Job Pipeline and tasks
+	JobPipeline *JobPipelineConfig
+
+	// Flux Monitor
+	FMDefaultTransactionQueueDepth *uint32
+	FMSimulateTransactions         *bool
+
+	// OCR V2
+	FeatureOffchainReporting2 *bool
+	OCR2                      *OCR2Config
+
+	// OCR V1
+	FeatureOffchainReporting *bool
+	OCR                      *OCRConfig
+
+	// P2P Networking
+	P2P *P2PConfig
+
+	// Keeper
+	Keeper *KeeperConfig
+
+	// Debugging
+	AutoPprof *AutoPprofConfig
+}
+
+type SecretsConfig struct {
+	DatabaseURL       *URL
+	ExplorerAccessKey string `toml:",omitempty"`
+	ExplorerSecret    string `toml:",omitempty"`
+	//TODO more?
```

Comment:
@samsondav Are secrets out of scope for the dump command?
https://app.shortcut.com/chainlinklabs/story/33624/add-secrets-toml